### PR TITLE
add function for casting ts as varchar

### PR DESCRIPTION
**Pull request by dongdongyang33**
Signed-off-by: wuqinxuan <wuqinxuan33@gmail.com>

**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2772

**What this PR does / why we need it:**

cast ts as varchar function

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by broccoliSpicy** [comment]
> Signed-off-by: wuqinxuan [wuqinxuan33@gmail.com](mailto:wuqinxuan33@gmail.com)
> 
> **What type of PR is this?**
> 
> * [ ]  API-change
> * [ ]  BUG
> * [ ]  Improvement
> * [ ]  Documentation
> * [ ]  Feature
> * [ ]  Test and CI
> * [ ]  Code Refactoring
> 
> **Which issue(s) this PR fixes:**
> 
> issue #
> 
> **What this PR does / why we need it:**
> 
> cast ts as varchar function
> 
> **Special notes for your reviewer:**
> 
> Not Available
> 
> **Additional documentation (e.g. design docs, usage docs, etc.):**
> 
> Not Available

link the issue number would be better

File: pkg/vectorize/typecast/typecast.go [hunk 1]
```diff
@@ -243,3 +244,15 @@ func UintToDecimal128[T constraints.Integer](xs []T, rs []types.Decimal128) ([]t
 func timestampToDatetime(xs []types.Timestamp, rs []types.Datetime) ([]types.Datetime, error) {
 	return types.TimestampToDatetime(xs, rs)
 }
+
+func timestampToVarchar(xs []types.Timestamp, rs *types.Bytes) (*types.Bytes, error) {
+	oldLen := uint32(0)
+	for i, x := range xs {
+		rs.Data = strconv.AppendInt(rs.Data, int64(x), 10)
```
[the remark this answers is not in the document]
Comment (dongdongyang33) [reply pkg/vectorize/typecast/typecast.go]: got it. thx u!
The cast function can't pass the precision parameter in so I use the default precision.

**Comment by broccoliSpicy** [comment]
> 

our ci works on utc timezone,

**Review by qingxinhome — APPROVED** [review]

**Comment by qingxinhome** [comment]
> Signed-off-by: wuqinxuan [wuqinxuan33@gmail.com](mailto:wuqinxuan33@gmail.com)
> 
> **What type of PR is this?**
> 
> * [ ]  API-change
> * [ ]  BUG
> * [ ]  Improvement
> * [ ]  Documentation
> * [x]  Feature
> * [ ]  Test and CI
> * [ ]  Code Refactoring
> 
> **Which issue(s) this PR fixes:**
> 
> issue #2772
> 
> **What this PR does / why we need it:**
> 
> cast ts as varchar function
> 
> **Special notes for your reviewer:**
> 
> Not Available
> 
> **Additional documentation (e.g. design docs, usage docs, etc.):**
> 
> Not Available

**Comment by qingxinhome** [comment]
I think you are missing the function registration

**Comment by dongdongyang33** [comment]
> 

fix.

**Comment by dongdongyang33** [comment]
> I think you are missing the function registration

fix.

**Pull request merged**